### ISSUE TITLE
Multi-line not filling in areas when area: true is set?

### ISSUE THREAD

**Issue by okhorseguy** [mozilla/metrics-graphics#252]
When using a single-line line chart with area: true set, I get a nicely filled in area.
If I'm using a multi-line chart (in this case, 4 lines), I get lines but no fill for any of the four.
I looked at some of your code as well as the css files and it looks as if (at first glance) you have
added fill colors for up to 5 lines/areas. 

Is there something I am missing that I need to do to enable the fill on a multi-line or is this a 
missing feature...bug...etc.?

Thanks!

**Comment by amiel** [comment]
It would be great to be able to opt in to using areas with multiline.

Would you be open to accepting a pull-request implementing one of the following:

1. Default `area` to `false` when there is more than one line.
    This way, it would be possible to opt-in to areas for multiline by setting `area: true` in the configuration.

2. Add a new option, `multiline_area`, for specifically opting in to the area feature for multiline graphs.

3. Allow enabling area for individual lines. I'm thinking the interface would be something like:

   ```javascript
   MG.data_graphic({
     area: [true, false, true],
     // or
     multiline_area: [true, false, true],
   })
   ```

Are any of these of interest to you?

I'd love to be able to use metrics-graphics without having to maintain a fork just for this one change.

**Comment by almossawi** [comment]
@amiel Apologies for the late reply.

Come to think of it, I actually wouldn't mind allowing ``area: true`` to override the default behavior for multi-lines. @hamilton?

**Comment by hamilton** [comment]
I have no problem with ``area: true`` overriding the default behavior, especially with the first API choice - ``area: [true, false, true]``, since this is generally consistent with how we're doing things, and adding an option for this separate from ``area`` would cause a small percentage of bloat. This means that ``area`` can either accept boolean (apply to all lines), or an array of booleans (apply one-by-one). We already do this for accessors, data, and others.

Not 100% sure what the use case is, though - seems like not-great information design to me, and would suggest against it. Could you describe your use case a little bit, or provide an example of why this is a good idea?

**Comment by amiel** [comment]
Hey @almossawi and @hamilton, thanks for the response.

We had brainstormed setting only one of our lines with area, but decided against it, opting for `area: true` with all lines. Since we don't actually have a use-case for the option to set `area` per line, I'm leaning towards the first option (multiline charts default to `area: false`, which can be overwritten).

For now, we're using [this branch](https://github.com/ActionSprout/metrics-graphics/commit/6ea63031c9fd0660417a5252ffa14ea425111b44), but I would love to be able to use the official bower package.

If you are interested, I'll drum up a pull-request.

**Comment by benschwarz** [comment]
@amiel, I'm interested. 
Given that there is a setting for `area: true/false`, it should probably add area fills. 
@hamilton was in favour of the change, so I think it'd be great to see it get through.

**Comment by benschwarz** [comment]
Line ref: https://github.com/mozilla/metrics-graphics/blob/master/src/js/charts/line.js#L600

**Comment by amiel** [comment]
Thanks for the input.

I started looking in to changing it so that `area` has a default based on other options, but can still be overridden when explicitly set. So far, I have [some tests :)](https://github.com/mozilla/metrics-graphics/compare/master...amiel:allow-multiline-areas).

However, here's what I found:

* `MG.data_graphic` sets the [default `area`](https://github.com/mozilla/metrics-graphics/blob/master/src/js/common/data_graphic.js#L92) very [early in the process](https://github.com/mozilla/metrics-graphics/blob/master/src/js/common/data_graphic.js#L123).

    It would be a pain to set the default here based on the number of lines, because `data` is not yet normalized.

* `args.data` gets normalized much [later in the process](https://github.com/mozilla/metrics-graphics/blob/master/src/js/charts/line.js#L755).

    We need the data to be normalized to pick a default for `area` so that we can easily determine if it is a multiline chart.

    However, after the data is normalized, a default for `area` has already been chosen and it is impossible to tell if the user supplied an `area` or not (ie, `typeof args.area === "undefined"` is always false at this point).

Possible Solutions
---------

There are two places I can see this logic living:

1. In the `init` process for setting defaults.

    In this case, we have the original `args` so we can check `typeof args.area === "undefined"`, but would need to introduce extra/duplicated logic to determine if this is going to be a multiline chart.

2. Somewhere in `charts/line.js`, after data has been normalized by `raw_data_transform`.

     In this case, we would need access to the original input args in order to check check `typeof args.area === "undefined"`.

     1. Is the original `args` passed in to [`MG.data_graphic`](https://github.com/mozilla/metrics-graphics/blob/master/src/js/common/data_graphic.js#L16) stored anywhere that could be checked?
     2. If not, would it make sense to explicitly store the original `area` before applying defaults?

         This would be something like:

         ```diff
        diff --git a/src/js/common/data_graphic.js b/src/js/common/data_graphic.js
        index bbe0266..4f4c023 100644
        --- a/src/js/common/data_graphic.js
        +++ b/src/js/common/data_graphic.js
        @@ -120,6 +120,7 @@ MG.data_graphic = function(args) {
           if (!args) { args = {}; }

           var selected_chart = MG.charts[args.chart_type || defaults.chart_type];
        +  args.original_area = args.area;
           merge_with_defaults(args, selected_chart.defaults, defaults);

           if (args.list) {
           ```

Is there any input on these solutions? or a possibility I've missed?

**Comment by amiel** [comment]
Ok, I went ahead and implemented Option 2 (see #750). It does the trick. Let me know what you think.

**Comment by sedwan** [comment]
+1
